docs/en/operations/settings/memory-overcommit.md

```diff
@@ -0,0 +1,31 @@
+# Memory overcommit
+
+Memory overcommit is an experimental technique intended to allow to set more flexible memory limits for queries.
+
+The idea of this technique is to introduce settings which can represent guaranteed memory amount of memory a query can use.
+When memory overcommit is enabled and memory limit is reached ClickHouse will select the most overcommit query and try to free memory by killing this query.
+
+When memory limit is reached any query will wait some time during atempt to allocate new memory.
+If selected query is killed and memory is freed within waiting timeout, query will continue execution after waiting, otherwise it'll be killed too.
```

Comment:
```suggestion
If selected query is killed and memory is freed within waiting timeout, the other queries will continue execution after waiting, otherwise, one of remaining queries will be killed too.
```

Reply:
Perhaps it's better
```suggestion
If timeout is passed and memory is freed, the query continues execution. Otherwise an exception will be thrown and the query is killed.
```